src/utils/markdown.rs

```diff
@@ -0,0 +1,243 @@
+//! Utils for working with the [MarkdownV2 message style.](https://core.telegram.org/bots/api#markdownv2-style)
+
+use std::string::String;
+
+/// Applies the bold font style to the string.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn bold(s: &str) -> String {
+    wrap(s, "*", "*")
+}
+
+/// Applies the italic font style to the string.
+/// Can be safely used with `utils::markdown::underline()`.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn italic(s: &str) -> String {
+    if s.starts_with("__") && s.ends_with("__") {
+        return wrap(&s[..s.len() - 1], "_", r"\r__");
+    }
+    wrap(s, "_", "_")
+}
+
+/// Applies the underline font style to the string.
+/// Can be safely used with `utils::markdown::italic()`.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn underline(s: &str) -> String {
+    // In case of ambiguity between italic and underline entities
+    // ‘__’ is always greadily treated from left to right as beginning or end of underline entity,
+    // so instead of ___italic underline___ we should use ___italic underline_\r__,
+    // where \r is a character with code 13, which will be ignored.
+    if s.starts_with("_") && s.ends_with("_") {
+        return wrap(s, "__", r"\r__");
+    }
+    wrap(s, "__", "__")
+}
+
+/// Applies the strikethrough font style to the string.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn strike(s: &str) -> String {
+    wrap(s, "~", "~")
+}
+
+/// Builds an inline link with an anchor.
+/// Escapes `)` and ``` characters inside the link url.
```

Comment:
```suggestion
/// Builds an inline link with an anchor.
///
/// Escapes `)` and ``` characters inside the link url.
```